clients/application-amqp/src/main/java/org/eclipse/hono/application/client/amqp/ProtonBasedApplicationClient.java

```diff
@@ -75,6 +81,29 @@ private String getTenantScopedLinkAddress(
         return String.format("%s/%s", endpointName, tenantId);
     }
 
+    @Override
+    public void addOnClientReadyHandler(final Handler<AsyncResult<Void>> handler) {
+        if (handler != null) {
+            lifecycleStatus.addOnStartedHandler(handler);
+        }
+    }
+
+    @Override
+    public Future<Void> start() {
```

Comment:
This method's behavior should at least be documented. In general, I do not like the fact that we are using the anti-pattern of delegating to a superclass method again here ...

Reply:
I've removed the `super.start()` invocation and adapted the javadoc.